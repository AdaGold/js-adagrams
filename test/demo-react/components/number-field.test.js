import React from 'react';
import NumberField from 'demo-react/components/number-field';
import { expectRenderToMatchSnapshot } from '../expect-render';
import { render } from 'ink-testing-library';

describe('NumberField', () => {
  it('renders correctly', () => {
    const jsx = <NumberField currentValue={ 5 }>Description of Field</NumberField>;
    const { lastFrame } = render(jsx);

    expect(lastFrame()).toMatchSnapshot();
  });

  it('renders correctly when active', () => {
    const jsx = <NumberField currentValue={ 5 } isActive>Description of Field</NumberField>;
    const { lastFrame } = render(jsx);

    expect(lastFrame()).toMatchSnapshot();
  });

  it('accepts input', () => {
    const jsx = <NumberField currentValue={ 5 } isActive >Description of Field</NumberField>;
    const { stdin, rerender, lastFrame } = render(jsx);
    rerender(jsx); // Hack: Force React/Ink to run useEffect handlers so the useInput callback gets registered.

    stdin.write('6');

    expect(lastFrame()).toContain('6');
  });

  it('ignores input when inactive', () => {
    const jsx = <NumberField currentValue={ 5 }>Description of Field</NumberField>;
    const { stdin, rerender, lastFrame } = render(jsx);
    rerender(jsx); // Hack: Force React/Ink to run useEffect handlers so the useInput callback gets registered.

    stdin.write('6');

    expect(lastFrame()).toContain('5');
  })
});