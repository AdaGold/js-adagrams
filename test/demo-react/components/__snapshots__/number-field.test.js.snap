// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberField renders correctly 1`] = `
"
  5                                                                             Description of Field
"
`;

exports[`NumberField renders correctly when active 1`] = `
"
  [7m5[27m                                                                             Description of Field
"
`;
